upf/upf.api

```diff
@@ -237,3 +237,67 @@ define upf_nwi_dump {
   u32 client_index;
   u32 context;
 };
+
+enum intf_type : u8 {
+  S_INTF_ACCESS = 0,
+  S_INTF_CORE = 1,
+  S_INTF_SGI_LAN = 2,
+  S_INTF_CP = 3,
+  S_INTF_INVALID = 255,
+};
+
+autoreply define upf_gtpu_endpoint_add_del {
+  u32 client_index;
+  u32 context;
+  u8 is_add;
+  vl_api_ip4_address_t ip4;
+  vl_api_ip6_address_t ip6;
+  u8 nwi[64];
+  u8 source_interface; /* [default=255] */
+  u32 teid;
+  u32 teidri;
+};
+
+define upf_gtpu_endpoint_details {
+  u32 client_index;
+  u32 context;
+  u32 teid;
+  u32 mask;
+  vl_api_ip4_address_t ip4;
+  vl_api_ip6_address_t ip6;
+  u8 nwi[64];
+  u8 source_interface; /* [default=255] */
+};
+
+define upf_gtpu_endpoint_dump {
+  u32 client_index;
+  u32 context;
+};
+
+autoreply define upf_pfcp_endpoint_add_del {
+  u32 client_index;
+  u32 context;
+  u8 is_add;
+  u32 table_id;
+  vl_api_address_t ip;
+};
+
+define upf_pfcp_endpoint_details {
+  u32 client_index;
+  u32 context;
+  u32 fib_table;
+  vl_api_address_t ip;
+};
+
+define upf_pfcp_endpoint_dump {
+  u32 client_index;
+  u32 context;
+};
+
+autoreply define upf_pfcp_server_set {
+  u32 client_index;
+  u32 context;
+  u32 fifo_size;  /* in KB */
+  u32 prealloc_fifos;
+  u32 segment_size; /* in MB */
+};
```

Comment:
It would be nice to have an API call for retrieving back the settings, otherwise, we will not be able to e.g. sync it in a k8s controller

Reply:
Done